### PR TITLE
CoreML documentation added for drawing classifier

### LINKED ISSUE #1964: Add end to end code for Core ML user guide
[The deployment for Core ML user guide page](https://apple.github.io/turicreate/docs/userguide/drawing_classifier/export-coreml.html) contains a lot of useful code. However it's not clear how to actually use a model given either stroke based data or images.

Let's have top level code give, which uses the code already on that page, that shows two things:
- [ ] Given a bmp file path, how to get predictions from a loaded model.
- [ ] Given a list of strokes, how to get predictions from a loaded model.

### PR DESCRIPTION
**Pull request by shreyajain17**
fixes #1964

### CODE REVIEW

File: userguide/drawing_classifier/export-coreml.md [hunk 1]
```diff
@@ -167,6 +162,39 @@ func rasterize(drawing stroke_based_drawing:Drawing) -> CGImage {
...
+
+
+Through a simple drag and drop process, you can incorporate the model into Xcode. The following Swift code is needed to consume the model in an iOS app.
+
+```swift
+let model = try VNCoreMLModel(for: MySquareTriangleClassifier().model)
+
+let request = VNCoreMLRequest(model: model, completionHandler: { [weak self] request, error in
+    self?.processClassifications(for: request, error: error)
+
```
Comment (TobyRoseman) [inline userguide/drawing_classifier/export-coreml.md]: You're missing a `})` here, right?

Even after adding that, I get the following error:
`Value of optional type 'VNCoreMLModel?' must be unwrapped to a value of type 'VNCoreMLModel'`.
This is blocking me from trying any more of the code here.

File: userguide/drawing_classifier/export-coreml.md [hunk 2]
```diff
@@ -167,6 +162,39 @@ func rasterize(drawing stroke_based_drawing:Drawing) -> CGImage {
...
+//Using Bitmap input
```
Comment (TobyRoseman) [inline userguide/drawing_classifier/export-coreml.md]: For this comment and the next one, you probably want a space after the `//`.

File: userguide/drawing_classifier/export-coreml.md [hunk 3]
```diff
@@ -167,6 +162,39 @@ func rasterize(drawing stroke_based_drawing:Drawing) -> CGImage {
...
+let grayscaleImage = UIImage(named: "MyImage")
```
Comment (TobyRoseman) [inline userguide/drawing_classifier/export-coreml.md]: Before looking at the UIKit documentation, it really wasn't clear to me that the `named` value is the filename.
I'd recommend changing this line to: `let grayscaleImage = UIImage(named: bitmapFilename)`

File: userguide/drawing_classifier/export-coreml.md [hunk 4]
```diff
@@ -21,20 +21,14 @@ Making a prediction at inference time on device is easy! In this section, we go
...
 the exported Core ML model. Note that the image you provide to the Vision API
 must be a Grayscale Image. 
 
-```swift
-let model = try VNCoreMLModel(for: MySquareTriangleClassifier().model)
-
-let request = VNCoreMLRequest(model: model, completionHandler: { [weak self] request, error in
-    self?.processClassifications(for: request, error: error)
-})
-```
```
Comment (nickjong) [inline userguide/drawing_classifier/export-coreml.md]: After looking at the final result here, I think it's too easy for someone glancing at this page to miss the fact that content "Using Stroke-Based Drawing Input" applies to the bitmap case here. Let's just have two copies of the Vision sample code, which will at least differ in how the input image is obtained.

File: userguide/drawing_classifier/export-coreml.md [hunk 5]
```diff
@@ -167,6 +162,39 @@ func rasterize(drawing stroke_based_drawing:Drawing) -> CGImage {
         space:grayscale, bitmapInfo:CGImageAlphaInfo.none.rawValue)
     let final_rect = CGRect(x: 0.0, y: 0.0, width: 28.0, height: 28.0)
     final_bitmap_context?.draw(intermediate_image!, in: final_rect)
-    return (final_bitmap_context?.makeImage())!
+    return cgImage : (final_bitmap_context?.makeImage())!
```
Comment (nickjong) [inline userguide/drawing_classifier/export-coreml.md]: Why do we need to add the `cgImage` annotation here?

File: userguide/drawing_classifier/export-coreml.md (hunk 2 above)
Comment (nickjong) [inline userguide/drawing_classifier/export-coreml.md]: Let's leave a space after the `//`

File: userguide/drawing_classifier/export-coreml.md [hunk 6]
```diff
@@ -167,6 +162,39 @@ func rasterize(drawing stroke_based_drawing:Drawing) -> CGImage {
...
+
+let request = VNCoreMLRequest(model: model, completionHandler: { [weak self] request, error in
+    self?.processClassifications(for: request, error: error)
+
+//Using Bitmap input
+let grayscaleImage = UIImage(named: "MyImage")
+let handler = VNImageRequestHandler(cgImage: grayscaleImage.cgImage!, options: [:])
+
+//Using stroke-based drawing as input
+let main_image : CGImage = rasterize(drawing: myDrawing)
```
Comment (nickjong) [inline userguide/drawing_classifier/export-coreml.md]: Let's omit the space between `main_image` and `:`

File: userguide/drawing_classifier/export-coreml.md [hunk 7]
```diff
@@ -167,6 +162,39 @@ func rasterize(drawing stroke_based_drawing:Drawing) -> CGImage {
...
+let model = try VNCoreMLModel(for: MySquareTriangleClassifier().model)
+
+let request = VNCoreMLRequest(model: model, completionHandler: { [weak self] request, error in
+    self?.processClassifications(for: request, error: error)
+
+//Using Bitmap input
+let grayscaleImage = UIImage(named: "MyImage")
+let handler = VNImageRequestHandler(cgImage: grayscaleImage.cgImage!, options: [:])
+
+//Using stroke-based drawing as input
```
Comment (nickjong) [inline userguide/drawing_classifier/export-coreml.md]: Yeah, if we're going to have to separate out the two cases via comments here, I think we may as well have two versions of this that are similar but have these differences.

File: userguide/drawing_classifier/export-coreml.md (hunk 6 above)
Comment (TobyRoseman) [inline userguide/drawing_classifier/export-coreml.md]: Looks like `myDrawing` is suppose to be of type `Drawing` (this type was defined earlier on this user guide page).

How do I create an instance of `Drawing` given raw stroke data?

Let's say I have this:
```swift
        let example_drawing = [
            [
                ["x": 1.0, "y": 2.0],
                ["x": 2.0, "y": 2.0],
                ["x": 3.0, "y": 2.0],
                ["x": 4.0, "y": 2.0],
                ["x": 5.0, "y": 2.0]
            ],
            [
                ["x": 10.0, "y": 10.0],
                ["x": 10.5, "y": 10.5],
                ["x": 11.0, "y": 11.0],
                ["x": 12.5, "y": 12.5],
                ["x": 15.0, "y": 15.0]
            ]
        ]
```
How do I create an instance of `Drawing` from this?

It's really not clear to me. Maybe the `init` function for drawing should take a list of list of dictionaries.

File: userguide/drawing_classifier/export-coreml.md (hunk 5 above)
Comment (shreyajain17) [reply userguide/drawing_classifier/export-coreml.md]: Left it by mistake

File: userguide/drawing_classifier/export-coreml.md [hunk 8]
```diff
@@ -29,7 +29,27 @@ the drawing you want to classify, you can use the Vision framework to consume
...
+let model = try VNCoreMLModel(for: MySquareTriangleClassifier().model)
+
+let request = VNCoreMLRequest(model: model, completionHandler: { [weak self] request, error in
+    self?.processClassifications(for: request, error: error)
+})
+
+let grayscaleImage = UIImage(named: bitmapFilename)
+let handler = VNImageRequestHandler(cgImage: grayscaleImage.cgImage!, options: [:])
+
+try? handler.perform([request])}
```
Comment (TobyRoseman) [inline userguide/drawing_classifier/export-coreml.md]: You don't want the `}`.

File: userguide/drawing_classifier/export-coreml.md [hunk 9]
```diff
@@ -29,7 +29,27 @@ the drawing you want to classify, you can use the Vision framework to consume
...
+
+func processClassifications(for request: VNRequest) {
```
Comment (TobyRoseman) [inline userguide/drawing_classifier/export-coreml.md]: I think you're missing an error parameter. I think this line should be:
`func processClassifications(for request: VNRequest, error: Error?) {`

Let's put this func in its own code block to make it clear that it's a separate function.

File: userguide/drawing_classifier/export-coreml.md [hunk 10]
```diff
@@ -29,7 +29,27 @@ the drawing you want to classify, you can use the Vision framework to consume
...
 
+```swift
+let model = try VNCoreMLModel(for: MySquareTriangleClassifier().model)
+
+let request = VNCoreMLRequest(model: model, completionHandler: { [weak self] request, error in
+    self?.processClassifications(for: request, error: error)
+})
+
+let grayscaleImage = UIImage(named: bitmapFilename)
+let handler = VNImageRequestHandler(cgImage: grayscaleImage.cgImage!, options: [:])
```
Comment (TobyRoseman) [inline userguide/drawing_classifier/export-coreml.md]: I had to `grayscaleImage` -> `grayscaleImage!` in order to get this to work.

File: userguide/drawing_classifier/export-coreml.md [hunk 11]
```diff
@@ -29,7 +29,27 @@ the drawing you want to classify, you can use the Vision framework to consume
...
+
+try? handler.perform([request])}
+
+func processClassifications(for request: VNRequest) {
+    if let sortedResults = request.results! as? [VNClassificationObservation] {
+        for result in sortedResults {
+            // Use results
```
Comment (TobyRoseman) [inline userguide/drawing_classifier/export-coreml.md]: Let's actually show how to do something here:
`print(result.identifier, result.confidence)`

File: userguide/drawing_classifier/export-coreml.md [hunk 12]
```diff
@@ -97,13 +117,43 @@ class Drawing {
...
+                ["x": 11.0, "y": 11.0],
+                ["x": 12.5, "y": 12.5],
+                ["x": 15.0, "y": 15.0]
+            ]
+        ]
+
+let myDrawing = Drawing()
+for stroke in example_drawing {
+    for point in stroke {
+        print(point)
```
Comment (TobyRoseman) [inline userguide/drawing_classifier/export-coreml.md]: Remove this print statement.

File: userguide/drawing_classifier/export-coreml.md [hunk 13]
```diff
@@ -162,25 +212,21 @@ func rasterize(drawing stroke_based_drawing:Drawing) -> CGImage {
...
-//Using Bitmap input
-let grayscaleImage = UIImage(named: "MyImage")
-let handler = VNImageRequestHandler(cgImage: grayscaleImage.cgImage!, options: [:])
-
-//Using stroke-based drawing as input
-let main_image : CGImage = rasterize(drawing: myDrawing)
+let main_image: CGImage = rasterize(drawing: myDrawing)
 let handler = VNImageRequestHandler(cgImage: main_image)        
 
 try? handler.perform([request])}
```
Comment (TobyRoseman) [inline userguide/drawing_classifier/export-coreml.md]: You don't want the `}`.

File: userguide/drawing_classifier/export-coreml.md [hunk 14]
```diff
@@ -29,7 +29,27 @@ the drawing you want to classify, you can use the Vision framework to consume
...
+
+try? handler.perform([request])}
+
+func processClassifications(for request: VNRequest) {
+    if let sortedResults = request.results! as? [VNClassificationObservation] {
+        for result in sortedResults {
+            // Use results
+        }
+    }
+}
```
Comment (TobyRoseman) [inline userguide/drawing_classifier/export-coreml.md]: Looks like this function is also used by the code in the stroke section. So I think you need to make these three changes there too.

File: userguide/drawing_classifier/export-coreml.md [hunk 15]
```diff
@@ -162,25 +212,21 @@ func rasterize(drawing stroke_based_drawing:Drawing) -> CGImage {
...
+The Core ML model can then run inference on this bitmap via the Vision framework
+as described above under "Using Bitmap Input".
 
 
-Through a simple drag and drop process, you can incorporate the model into Xcode. The following Swift code is needed to consume the model in an iOS app.
-
 ```swift
 let model = try VNCoreMLModel(for: MySquareTriangleClassifier().model)
 
 let request = VNCoreMLRequest(model: model, completionHandler: { [weak self] request, error in
```
Comment (TobyRoseman) [inline userguide/drawing_classifier/export-coreml.md]: I think you'll need to `model: model` -> `model: model!`

**Pull request merged**